zebra-network/src/address_book.rs

```diff
@@ -463,6 +514,13 @@ impl AddressBook {
         self.by_addr.descending_values().cloned()
     }
 
+    /// Returns true if there is a recently live peer connection with the provided IP address.
+    fn has_active_peer_with_ip(&self, ip: &IpAddr, chrono_now: chrono::DateTime<Utc>) -> bool {
+        self.most_recent_by_ip
+            .get(ip)
+            .map_or(false, |peer| peer.was_recently_live(chrono_now))
```

Comment:
Note: `last_response` times can remain live for a long time if the local clock is changed to a much earlier time. There's not much we can do about that, but it might be worth documenting, since checking by IP makes the problem slightly worse.

(It impacts all nodes with that IP, not just the node with that exact IP and port.)

Reply:
Documented in https://github.com/ZcashFoundation/zebra/pull/7029/commits/0c769b657fbe1f66dd20b7ed874ce4b529395395.

Reply:
Should it be updated to make `last_response` an `Instant`, or a tuple with both?

Reply:
> Should it be updated to make `last_response` an `Instant`, or a tuple with both?

I think that's beyond the scope of a small security fix?

(We need to have a clock time because responded times are sent as part of the Zcash network protocol.)